backend/cfg/cfg_with_layout.ml

```diff
@@ -388,6 +388,15 @@ let reorder_blocks_random ?random_state t =
   in
   set_layout t (DLL.of_list new_layout)
 
+let reorder_blocks ~comparator t =
+  (* Ensure entry block remains first *)
+  let original_layout = DLL.to_list (layout t) in
```

Comment:
I don't want to block this pull request,
so that should probably be a mere CR:
we do not need a full-blown sort here,
and we could eventually avoid the
conversion to list and back.

Reply:
(Sorry, that comment does not really
make sense here: it makes slightly more
sense if we consider the only sorting
criterion we use -at this point- is a
boolean field.)

Reply:
Ah, that would be possible indeed. Maybe it would just be better to implement sort on `DLL`? I think merge sort is stable, and is quite adapted to DLLs. I'll add a CR.

Reply:
Sure, thanks.